multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/UploadAppStep.java

```diff
@@ -94,6 +99,34 @@ public StepPhase executeAsyncStep(ProcessContext context) throws FileStorageExce
         return StepPhase.DONE;
     }
 
+    private CloudPackage createDockerPackage(CloudControllerClient client, CloudApplicationExtended application) {
+        UUID applicationGuid = client.getApplicationGuid(application.getName());
+        return client.createDockerPackage(applicationGuid, application.getDockerInfo());
+    }
+
+    private String getNewApplicationDigest(ProcessContext context, String appArchiveId, String fileName) throws FileStorageException {
+        return fileService.processFileContent(context.getVariable(Variables.SPACE_GUID), appArchiveId,
```

Comment:
Can `appArchiveId` be retrieved in this method instead of passing it as a parameter?

Reply:
Yes, it can.